### PR TITLE
[Feature] Add a scheduling strategy for high-throughput point query scenarios.

### LINKED ISSUE #7066: [Feature] Support config scheduling strategy for scannode.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

When schedule ScanNode, the current strategy will schedule scannodes to as many backends as possible, which can efficiently utilize cluster resources and improve query performance.

However, for high throughput point query scenarios, the current scheduling strategy has some problems
1. Point query itself does not require many resources, so it is of little significance to mobilize all backends to serve query.
2. The thundering herd will be serious, which will affect the query performance and query parallelism of the cluster
3. It will increase the RPC and data interaction between backends, which will have a great impact on the performance

Therefore, maybe we should support a new scheduling strategy for high-throughput point query scenarios, select as few backends as possible to serve query.

### Use case

Suppose you have a table with many partitions, and the query scenario is high-throughput point query (each query will specify the distribute key)

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by killxdcj**
## Proposed changes

Support config scheduling strategy. More detail in: #7066

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #7066) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

More detail in: #7066

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 1]
```diff
@@ -384,6 +386,9 @@
     @VariableMgr.VarAttr(name = ENABLE_LATERAL_VIEW, needForward = true)
     public boolean enableLateralView = false;
 
+    @VariableMgr.VarAttr(name = EXEC_SCHEDULING_STRATEGY)
+    public String execSchedulingStrategy = ExecSchedulingStrategy.EVENLY_BACKEND.name();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:390]: For `String` type session variable, user may set any unexpected values.
I think you refer `sql_mode` or `runtime_filter_type`.
And also refer to the `VariableVarConverters.java`

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 2]
```diff
@@ -1456,6 +1463,47 @@ private void computeScanRangeAssignmentByScheduler(
         }
     }
 
+    private void computeScanRangeAssignmentWithMinNodeStrategy(
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:1466]: Could you please add some comment to explain the logic of this method?

Comment (killxdcj) [reply fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:1500]: comments resolved

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java (hunk 1 above)
Comment (killxdcj) [reply fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java]: Resolved.

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 3]
```diff
@@ -1359,7 +1360,13 @@ private void computeScanRangeAssignment() throws Exception {
                 bucketShuffleJoinController.computeScanRangeAssignmentByBucket((OlapScanNode) scanNode, idToBackend, addressToBackendID);
             }
             if (!(fragmentContainsColocateJoin | fragmentContainsBucketShuffleJoin)) {
-                computeScanRangeAssignmentByScheduler(scanNode, locations, assignment, assignedBytesPerHost);
+                long schedulingStrategy = (ConnectContext.get() == null ? ExecSchedulingStrategy.EVENLY_BACKEND.getValue() :
+                    ConnectContext.get().getSessionVariable().getExecSchedulingStrategy());
+                if (schedulingStrategy == ExecSchedulingStrategy.MIN_BACKEND.getValue()) {
+                    computeScanRangeAssignmentWithMinNodeStrategy(scanNode, locations, assignment, assignedBytesPerHost);
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:1366]: could you supplement UT for this new strategy?

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 4]
```diff
@@ -154,17 +154,19 @@
     public static final long DEFAULT_INSERT_VISIBLE_TIMEOUT_MS = 10_000;
 
     public static final String EXTRACT_WIDE_RANGE_EXPR = "extract_wide_range_expr";
-
```
Comment (Gabriel39) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:157]: unnecessary update?

**Review by Gabriel39 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**